packages/shopping/src/controllers/user-management.controller.ts

```diff
@@ -297,4 +306,70 @@ export class UserManagementController {
 
     return {token};
   }
+
+  @post('/users/reset-password/init', {
+    responses: {
+      '200': {
+        description: 'Confirmation that reset password email has been sent',
+      },
+    },
+  })
+  async resetPasswordInit(
+    @requestBody() resetPasswordInit: ResetPasswordInit,
+  ): Promise<string> {
+    if (!isemail.validate(resetPasswordInit.email)) {
+      throw new HttpErrors.UnprocessableEntity('Invalid email address');
+    }
+
+    const nodeMailer: NodeMailer = await this.userManagementService.requestPasswordReset(
+      resetPasswordInit.email,
+    );
+
+    if (nodeMailer.accepted.length) {
+      return 'Successfully sent reset password link';
+    }
+    throw new HttpErrors.InternalServerError(
+      'Error sending reset password email',
+    );
+  }
+
+  @put('/users/reset-password/finish', {
+    responses: {
+      '200': {
+        description: 'A successful password reset response',
+      },
+    },
+  })
+  async resetPasswordFinish(
+    @requestBody() keyAndPassword: KeyAndPassword,
+  ): Promise<string> {
+    validateKeyPassword(keyAndPassword);
+
+    const foundUser = await this.userRepository.findOne({
+      where: {resetKey: keyAndPassword.resetKey},
+    });
```

Comment:
We might want to consider putting a time limit (e.g. 24 h) for the reset key to avoid a scenario where a user may have a password reset link with an "unlimited lifetime".

Reply:
Yes. That would involve adding another property `passwordResetRequestTime` or something and then checking the time that has passed since it was requested, when the reset URL is called.

Reply:
Included

Reply:
@achrinza please confirm if you are happy with these changes